yazi-core/src/files/files.rs

```diff
@@ -322,6 +324,44 @@ impl Files {
 		self.sorter.sort(&mut self.items, &self.sizes)
 	}
 
+	// --- Filter
+	pub fn set_filter(&mut self, keyword: &str) -> bool {
+		if keyword == self.filter.as_deref().unwrap_or_default() {
+			return false;
+		}
+
+		let old_hidden: Vec<File>;
+		(self.hidden, old_hidden) =
+			self.hidden.iter().cloned().partition(|f| !self.show_hidden && f.is_hidden());
```

Comment:
Is this should be `self.show_hidden` not `!self.show_hidden`?

Reply:
Perhaps this should be 2 separate lines, so it doesn't need to keep checking `self.show_hidden`, but I thought that logic was correct. Maybe I'm misunderstanding but shouldn't it go =>

1. `show_hidden = true` means all files are put in `old_hidden` so they can be re-added to `self.items`
2. `show_hidden = false` means any files that should be hidden (`f.is_hidden()`) remain in `self.hidden` but everything else goes to `old_hidden`

Reply:
Sorry, after some thinking, I still can't understand it.

Could you break it down into multiple lines? - I'll try to refactor it once I understand. 🤣

Reply:
Sure yeah

Reply:
Not sure why that formatting is weirdly unindented by 1 level but the change should be in that latest commit

Reply:
How are you doing formatting by the way? Just specific chunks, using that `rustfmt.toml`, rather than the whole file?

Reply:
`rustfmt.toml` for Yazi used some nightly features, so it requires `rustfmt` nightly.

You can do a `rustfmt +nightly **/*.rs` to format all files (might need to install rustfmt nightly first via `rustup`).

Reply:
Ah ok thanks